src/vm/checker/typecheck/tests/mod.rs

```diff
@@ -222,3 +222,100 @@ fn test_tuple_map() {
 
     type_check(&":transient:", &mut t, &mut analysis_db, false).unwrap();
 }
+
+
+#[test]
+fn test_explicit_tuple_map() {
```

Comment:
We should have some tests that _fail_ type validation as well -- e.g., they call

`(fetch-entry kv-store ((keys 0)))`

Which should fail, since `keys` isn't a valid entry in a tuple of `kv-store`.